### PR TITLE
Format charlists, fixes compilation warnings on Elixir 1.17

### PR DESCRIPTION
**Pull request by sabiwara**
Screenshot 2024-05-26 at 12 42 43

### CODE REVIEW

File: lib/codegen.ex [hunk 1]
```diff
@@ -48,14 +48,15 @@ defmodule Jason.Codegen do
...
       |> Enum.intersperse(",")
 
-    collapse_static(List.flatten(["{", elements] ++ '}'))
+    collapse_static(List.flatten(["{", elements] ++ ~c'}'))
   end
 
-  defp clauses_to_ranges([{:->, _, [[{:in, _, [byte, range]}, rest], action]} | tail], acc) do
-    clauses_to_ranges(tail, [{range, {byte, rest, action}} | acc])
+  defp clauses_to_ranges([{:->, _, [[{:in, _, [byte, range]}, rest], action]} | tail], acc, env) do
+    range = Macro.expand(range, env)
```
Comment (sabiwara) [inline lib/codegen.ex:55]: Unlike charlists, sigils are not their own AST, so we need to expand them first.

**Comment by michalmuskala** [comment]
Thank you! ❤️

**Comment by odarriba** [comment]
Any release coming soon (a patch version at least)? Thanks <3

**Comment by michalmuskala** [comment]
This was release in 1.4.2 (and 1.4.3) last week

**Pull request merged**